src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ImmutableDispatchRuntime.cs

```diff
@@ -640,6 +667,44 @@ internal async Task<MessageRpc> ProcessMessageAsync(MessageRpc rpc)
             return rpc;
         }
 
+        private (bool isConfigured, HttpContext httpContext, string authenticationScheme) IsCustomAuthenticationSchemeConfigured()
+            =>  OperationContext.Current.IncomingMessageProperties.TryGetValue<HttpContext>(
+                    "Microsoft.AspNetCore.Http.HttpContext", out var httpContext)
+                && httpContext.Items.TryGetValue("CoreWCF.Channels.HttpTransportSettings.CustomAuthenticationScheme",
+                    out var authenticationScheme)
+                && authenticationScheme is string authenticationSchemeValue
+                && !string.IsNullOrEmpty(authenticationSchemeValue)
+                ? (true, httpContext, authenticationSchemeValue)
+                : (false, null, null);
+
+        private async ValueTask<(bool isAuthorized, Message message)> EvaluateAuthorizationPolicies(MessageRpc rpc)
+        {
+            var httpContext = OperationContext.Current.IncomingMessageProperties["Microsoft.AspNetCore.Http.HttpContext"] as HttpContext;
+            string authenticationScheme = httpContext.Items["CoreWCF.Channels.HttpTransportSettings.CustomAuthenticationScheme"] as string;
+            var principal = httpContext.User;
+            AuthorizationOperationInvoker invoker = (AuthorizationOperationInvoker)rpc.Operation.Invoker;
+
```

Comment:
You can't depend on the operation invoker being a specific type. Just as you wrapped the existing invoker with your own, a developer might do the same. You are using the invoker as a means to hold properties about the operation, but that's the wrong place to store that. I had a look at the code that uses the asp.net core policy implementation and the same set of attributes are universally used by gRPC, WebAPI, MVC, and SignalR so I think it's a good idea to build in to the core code of CoreWCF knowledge about the concept of policies. I believe we should have policies also work for non-http transports too so we're not adding http specific implementation in primitives.   

I would make a public api on DispatchOperation which holds a collection of IAuthorizeData. This would allow attributing (which CoreWCF would populate), or a custom behavior which populates it more directly (for example, might not want to populate it the same for unit tests or dev environment).

Reply:
done. please see the comment above.

Reply:
done. please see the comment above.